src/wallet/rpcwallet.cpp

```diff
@@ -426,6 +427,14 @@ UniValue claimname(const UniValue& params, bool fHelp)
         );
     string sName = params[0].get_str();
     string sValue = params[1].get_str();
+
+    if (sName.empty())
+        throw JSONRPCError(RPC_WALLET_ERROR, "Claim name should not be empty");
+
```

Comment:
while it is unintuitive, an empty claim name is actually valid (you can make claim for empty names when you use double quotes: "" in the RPC claimname command) so this check is not needed